lib/spack/env/cc

```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```

Comment:
This might be the only place I hesitate to use `/usr/bin/env sh`. How much overhead does this result in? Build scripts call `cc` thousands of times, so if we have to search for `bash` in the `PATH` every time that could be slow.

Reply:
maybe we should bootstrap a static binary executable to replace the bash script ^^. i was meaning to check what the overhead of the shell wrapper actually is

Reply:
This might actually help us on windows -- where there is no `bash`.  If you get numbers on this I'd be interested.  But I also worry b/c ultimately not every Spack build will actually require a compiler, so we may not have one handy (unlikely, I know).  But sure we could use the bootstrap logic and also provide binaries for something this simple.

On a similar note: we have talked about (for a while) generating compiler wrappers for each build, to avoid the symlink farm currently in `lib/spack/env` and the lengthy switch statement in `lib/spack/env/cc`.  Then the names of the compilers would be completely up to the classes in `lib/spack/spack/compilers`, or, ultimately, to the compiler packages (when compilers become normal dependencies).